beacon-chain/blockchain/service.go

```diff
@@ -36,10 +36,9 @@ type ChainFeeds interface {
 	StateInitializedFeed() *event.Feed
 }
 
-// NewHeadNotifier defines a struct which can notify many consumers of a new,
-// canonical chain head event occuring in the node.
-type NewHeadNotifier interface {
-	HeadUpdatedFeed() *event.Feed
+// StateFeeder interface defines the methods of the service that provides state updates to consumers.
+type StateFeeder interface {
+	StateFeed() *event.Feed
```

Comment:
Name is bit vague, how about `StateUpdatedFeed()` or `StateProcessedFeed()` ?

Reply:
I wanted to keep it vague to give more freedom to the events that could be put in here.  For example, at some stage it may be useful to have a timer that ticks at the start of each epoch but that's not really a specific update or process.

Reply:
Went with `StateNotifier` as per our discussion.